pkg/cli/jobserver/commands/serve.go

```diff
@@ -0,0 +1,89 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+
+	rbacv1 "github.com/kralicky/jobserver/pkg/apis/rbac/v1"
+	"github.com/kralicky/jobserver/pkg/auth"
+	"github.com/kralicky/jobserver/pkg/cgroups"
+	"github.com/kralicky/jobserver/pkg/jobs"
+	"github.com/kralicky/jobserver/pkg/rbac"
+	"github.com/kralicky/jobserver/pkg/server"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/encoding/protojson"
+	"sigs.k8s.io/yaml"
+)
+
+// ServeCmd represents the serve command
+func BuildServeCmd() *cobra.Command {
+	var rbacConfigFile string
+	var serverConfig server.Options
+	cmd := &cobra.Command{
+		Use:   "serve",
+		Short: "A brief description of your command",
+		Long:  ``,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if os.Geteuid() != 0 {
+				return fmt.Errorf("server must be run as root (try 'sudo %s serve')", os.Args[0])
+			}
```

Comment:
Not really convinced that this is necessary, especially with a fixed cgroup path; you can just set that up to be writable by the current user and then you don't need to be privileged.

Reply:
Does that have security implications? We could also have the server set the user's uid/gid on processes too, although it was mentioned previously that that was out of scope.

Reply:
cgroups v2 have very powerful and simple delegation properties, you can give a user the ability to create sub-cgroups and configure controllers in a cgroup, with total limits enforced on the root of the delegated hierarchy that the user can't modify, and it's all done by assigning file permissions to the appropriate files and directories. I think that you might be restricted from moving processes across cgroups if you don't have write permissions over a whole chain of cgroups from source to destination, but that's also easily solved by starting the jobserver in `/jobserver-kralicky/jobserver` and then letting it move processes into `/jobserver-kralicky/<uuid>`.

Reply:
This is not something that the program should do, it's just something that the program shouldn't unnecessarily restrict. If the cgroup hierarchy isn't already set up in such a way that the user can manipulate the cgroups as necessary, it'll just error out with a permission error, and that's fine.

Reply:
Gotcha, that makes sense. I agree that the restriction does seems arbitrary, so I've removed it. :+1: